### PR TITLE
Fix issue #103 - only include analytics.js on certain pages

### LINKED ISSUE #103: Only load js/analytics.js on thank you pages
Currently, js/analytics.js is loaded on all pages.

But we only need to call this on
`/thank-you*`
and
`/share*`
pages

This results in one less http request, and a few less regex expressions to run on the rest of our forms.

Ideally, we could mark a page as being of type 'thank-you' so that if we create multiple thank you pages for testing, we don't forget to add this tracking in the future.

### PR DESCRIPTION
**Pull request by alicoding**
closes #103

### CODE REVIEW

**Comment by alicoding** [comment]
@ScottDowne R?

File: pages/share.jsx [hunk 1]
```diff
@@ -2,9 +2,13 @@ import React  from 'react';
 import Footer from '../components/footer.jsx';
 import Header from '../components/header.jsx';
 import Social from '../components/social.jsx';
+import analytics from '../public/js/analytics.js';
 
 var ThankYou = React.createClass({
   mixins: [require('react-intl').IntlMixin],
+  componentDidMount: function() {
+    analytics();
```
Comment (ScottDowne) [inline pages/share.jsx]: I am pretty sure we don't want this here.

1. It'll double count locales with share in the thank you page.
2. Share itself isn't a page that has transaction data we need to track.

**Comment by ScottDowne** [comment]
Just the one comment.

File: pages/share.jsx (hunk 1 above)
Comment (alicoding) [inline pages/share.jsx]: Based on this https://github.com/mozilla/donate.mozilla.org/issues/103 not sure if I misunderstood something from there?

Comment (ScottDowne) [inline pages/share.jsx]: Yeah, and share is not the thank you page.

Comment (ScottDowne) [inline pages/share.jsx]: Oh, I see, adam is asking explicitly for share to have it... odd.

**Pull request merged**